src/hardware/ad7172.rs

```diff
@@ -176,11 +176,12 @@ impl Ad7172 {
         // 5000 us delay after reset.
         delay.delay_us(5000u16);
 
-        let id = adc.read(AdcReg::ID);
-        // check that ID is 0x00DX, as per datasheet
-        if id & 0xf0 != 0x00d0 {
-            return Err(Error::AdcId);
-        }
+        // TODO move that
```

Comment:
Yes. Move all this (reset to ID check) into a `setup()`. You need to call it once per ADC.